External/pubchem_shape/PubChemShape.hpp

```diff
@@ -2,11 +2,47 @@
 #include <map>
 #include <vector>
 
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+#include <boost/serialization/map.hpp>
+#include <boost/serialization/vector.hpp>
+
 #ifndef RDKIT_PUBCHEMSHAPE_GUARD
 #define RDKIT_PUBCHEMSHAPE_GUARD
 
 //! The input for the pubchem shape alignment code
 struct RDKIT_PUBCHEMSHAPE_EXPORT ShapeInput {
+  ShapeInput() = default;
+  ShapeInput(const std::string &str) {
+    std::stringstream ss(str);
+    boost::archive::text_iarchive ia(ss);
+    ia &*this;
```

Comment:
```suggestion
    ia & *this;
```

Reply:
clang-format takes out the space, on both my Ubuntu box and macs.

Reply:
huh... that's unlucky

Reply:
Yeah.  It looks ugly.  I figure we live with it, or it will be re-formatted every time someone changes the file, but I can always tweak it in emacs for now and re-push.

Reply:
Na, leave it